src/parser/Lexer.cpp

```diff
@@ -1894,7 +1900,18 @@ void Scanner::lex(Scanner::ScannerResult* token)
         return;
     }
 
-    const char16_t cp = this->peekCharWithoutEOF();
+    char16_t cp = this->peekCharWithoutEOF();
+    
+    if (UNLIKELY(this->peekChar() >= 0xD800 && this->peekChar() < 0xDFFF)) {
+        cp = peekChar();
+        ++this->index;
+        char32_t ch2 = this->peekChar();
+        if (U16_IS_TRAIL(ch2)) {
+            cp = U16_GET_SUPPLEMENTARY(cp, ch2);
+        } else {
+            this->throwUnexpectedToken();
+        }
+    }
```

Comment:
IMO this part can be removed because `isIdentifierPartStart` would recognize identifier correctly once unicode table is updated to the latest version.

Reply:
Atm this code is needed for the character that are in the Supplementary Range. For ex.:
```js
var 𑴀 = "foo"
print(𑴀)
```
will fail with ILLEGAL TOKEN error.